### PR TITLE
Create tee attributes.md

### PR DESCRIPTION
**Pull request by spjohnso**
Created file in response to https://github.com/confidential-computing/governance/issues/26

### CODE REVIEW

File: tee attributes.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+## TEE Attributes
+
+Required properties to be called a “TEE”:
+* ___Data integrity___ - data values stored or being operated upon within the bounds of the TEE cannot be modified by an actor operating outside the TCB.
+* ___Data confidentiality___ - data values stored or being operated upon within the bounds of the TEE cannot be determined by an actor operating outside the TCB.
+* ___Code integrity___ - software codes executed inside the bounds of the TEE cannot be modified by an actor operating outside the TCB.
+
+TEE's may have other properties:
+* _Code confidentiality_ - software codes executed inside the bounds of the TEE cannot be modified by an actor operating outside the TCB.
```
Comment (dthaler) [inline tee attributes.md:9]: Looks like a cut and paste error from line 6.  The definition should be different.
```suggestion
* _Code confidentiality_ - software codes executed inside the bounds of the TEE cannot be determined by an actor operating outside the TCB.
```

File: tee attributes.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+* _Programmability_ - the extent to which the software codes operated upon inside the bounds of the TEE can be provided by an identified actor.
```
Comment (dthaler) [inline tee attributes.md:10]: "can be provided" I think should be explicitly "after time of manufacture" or something similar.

File: tee attributes.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+* _Attestability_ - the mechanism for demonstrating the contents of a TEE is infact secured by a TEE.
```
Comment (dthaler) [inline tee attributes.md:11]:
```suggestion
* _Attestability_ - the mechanism for demonstrating the contents of a TEE is in fact secured by a TEE.
```

Comment (dthaler) [reply tee attributes.md:11]: I'd also compare the text currently in the whitepaper, which might have a better definition

File: tee attributes.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+* _Unspoofability (un-clonable) / non-repudiability)_ - when using cryptographic authentication mechanisms in such a manner as it difficult to deny the origin of an assertion (often in referring to identity tied to a transaction).
+* _Authenticated launch_ - (aka secure boot) a mechanism for determining that an approved TCB via cryptographic signature is established. As opposed to “measured” launch which just establishes which elements compose the TCB without determination of its origin or acceptability.
+* _Recoverability_ if an error in the TCB of the TEE is discovered the ability for the error to be mitigated. Full recoverability allows for the attestation of the recovered TCB.
```
Comment (dthaler) [inline tee attributes.md:14]:
```suggestion
* _Recoverability_ - if an error in the TCB of the TEE is discovered the ability for the error to be mitigated. Full recoverability allows for the attestation of the recovered TCB.
```

File: tee attributes.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+* ___Code integrity___ - software codes executed inside the bounds of the TEE cannot be modified by an actor operating outside the TCB.
+
+TEE's may have other properties:
+* _Code confidentiality_ - software codes executed inside the bounds of the TEE cannot be modified by an actor operating outside the TCB.
+* _Programmability_ - the extent to which the software codes operated upon inside the bounds of the TEE can be provided by an identified actor.
+* _Attestability_ - the mechanism for demonstrating the contents of a TEE is infact secured by a TEE.
+* _Unspoofability (un-clonable) / non-repudiability)_ - when using cryptographic authentication mechanisms in such a manner as it difficult to deny the origin of an assertion (often in referring to identity tied to a transaction).
+* _Authenticated launch_ - (aka secure boot) a mechanism for determining that an approved TCB via cryptographic signature is established. As opposed to “measured” launch which just establishes which elements compose the TCB without determination of its origin or acceptability.
```
Comment (MikeCamel) [inline tee attributes.md:13]: Do we mean of code _within_ or _external to_ the TEE?

**Comment by MikeCamel** [comment]
I wonder whether we might add something like this added (as recoverability isn't always plausible):
_Tamper evidence_ - a mechanism to inform an interested party (either within or external to the TEE) that tampering was attempted or successful by an unauthorised party.

File: tee attributes.md (hunk 5 above)
Comment (spjohnso) [reply tee attributes.md:13]: inside <--> within

File: tee attributes.md (hunk 2 above)
Comment (spjohnso) [reply tee attributes.md:10]:
```suggestion
* _Programmability_ - the extent to which the software codes operated upon inside the bounds of the TEE provided after time of manufacturer by an identified actor.
```

**Comment by spjohnso** [comment]
Is tamper evidence a physical property of the TEE or the device that hosts the TEE? I am not sure you can reliably demonstrate tamper evidence remotely. Do you have any examples of that?

File: tee attributes.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+## TEE Attributes
+
+Required properties to be called a “TEE”:
+* ___Data integrity___ - data values stored or being operated upon within the bounds of the TEE cannot be modified by an actor operating outside the TCB.
+* ___Data confidentiality___ - data values stored or being operated upon within the bounds of the TEE cannot be determined by an actor operating outside the TCB.
+* ___Code integrity___ - software codes executed inside the bounds of the TEE cannot be modified by an actor operating outside the TCB.
+
+TEE's may have other properties:
+* _Code confidentiality_ - software codes executed inside the bounds of the TEE cannot be determined by an actor operating outside the TCB.
+* _Programmability_ - the extent to which the software codes operated upon inside the bounds of the TEE provided after time of manufacturer by an identified actor.
```
Comment (dthaler) [inline tee attributes.md:10]: The others above are phrased as a sentence.   This one is phrased as a definition.  I'd recommend picking one style and being consistent.

File: tee attributes.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+* _Attestability_ - the mechanism for demonstrating the contents of a TEE is in fact secured by a TEE.
```
Comment (dthaler) [inline tee attributes.md:11]: Grammar: "contents ... is" doesn't seem to match

File: tee attributes.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+* _Unspoofability (un-clonable) / non-repudiability)_ - when using cryptographic authentication mechanisms in such a manner as it difficult to deny the origin of an assertion (often in referring to identity tied to a transaction).
+* _Authenticated launch_ - (aka secure boot) a mechanism for determining that an approved TCB via cryptographic signature is established. As opposed to “measured” launch which just establishes which elements compose the TCB without determination of its origin or acceptability.
+* _Recoverability_ - if an error in the TCB of the TEE is discovered the ability for the error to be mitigated. Full recoverability allows for the attestation of the recovered TCB.
+* _Isolation_ - A mechanism by which the main properties of data & code integrity can be achieved.
```
Comment (dthaler) [inline tee attributes.md:15]: The others are properties, this seems to be phrased as a mechanism, not a property

File: tee attributes.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+* _Robustness_ - Actors operating outside the TCB may be defined as a software actor (e.g. the host operating system) or a hardware actor (e.g. the platform operator). The extent to which a TEE provides protection from one actor or another is defined by its TCB.
```
Comment (dthaler) [inline tee attributes.md:16]: I'm having a hard time understanding this one.

**Comment by dthaler** [comment]
> Is tamper evidence a physical property of the TEE or the device that hosts the TEE? I am not sure you can reliably demonstrate tamper evidence remotely. Do you have any examples of that?

Here's one academic paper on the topic from 2003: http://www.scs.stanford.edu/nyu/05sp/sched/readings/aegis.pdf

**Comment by spjohnso** [comment]
> Here's one academic paper on the topic from 2003: http://www.scs.stanford.edu/nyu/05sp/sched/readings/aegis.pdf

I see. One could argue that the scheme in this paper is just a form external memory integrity. And isn't fully a tamper detection or evidence, because there may be other things I can tamper with that the memory scheme is not protecting against.

**Comment by dthaler** [comment]
@spjohnso Now that we have the TAC whitepaper on the website, do we still need this PR or can we consider it overtaken by the whitepaper?

**Pull request closed without merging**